pkg/util/pem/pem_test.go

```diff
@@ -0,0 +1,45 @@
+package pem
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"testing"
+
+	utiltls "github.com/Azure/ARO-RP/pkg/util/tls"
+)
+
+func TestEncode(t *testing.T) {
```

Comment:
It looks like this one function has multiple given/when/then test cases centered around `Encode` calls. Can we break out each of these into named `t.Run` clauses to give some context for each, which would have added benefit to adding context to each `t.Fatal` (which, as it stands now, would be hard to know at which point something went wrong)